### PR TITLE
Update mailmap to consolidate contributions 

### PR DESCRIPTION
**Pull request by GiudGiud**
refs #22378

### CODE REVIEW

**Comment by GiudGiud** [comment]
there are some weird lines because people have messed up their email in their git config. 
most lines are people using different emails at various times or on various machines

**Comment by GiudGiud** [comment]
Failures unrelated

File: .mailmap [hunk 1]
```diff
@@ -112,3 +122,32 @@ Casey Icenhour <casey.icenhour@inl.gov>
 Guillaume Giudicelli <guillaume.giudicelli@inl.gov>
 Guillaume Giudicelli <guillaume.giudicelli@gmail.com>
 Guillaume Giudicelli <g_giud@mit.edu>
+Nathaniel Peats <76120892+socratesgorilla@users.noreply.github.com>
+Nathaniel Peats <nathaniel.peat@inl.gov>
```
Comment (loganharbour) [inline .mailmap:126]:
```suggestion
Nathaniel Peat <76120892+socratesgorilla@users.noreply.github.com>
Nathaniel Peat <nathaniel.peat@inl.gov>
```

**Review by loganharbour — CHANGES_REQUESTED** [review]

File: .mailmap (hunk 1 above)
Comment (GiudGiud) [reply .mailmap]: good catch

**Review by loganharbour — APPROVED** [review]

**Comment by loganharbour** [comment]
Unrelated failures.

**Pull request merged**